plugins/imjournal/imjournal.c

```diff
@@ -615,6 +673,9 @@ loadJournalState(void)
 	int r;
 	FILE *r_sf;
 
+	DBGPRINTF("Loading journal position, at head? %d, reloaded? %d\n",
+			  journalContext.atHead, journalContext.reloaded);
+
 	if (cs.stateFile[0] != '/') {
```

Comment:
This seems now to be called multiple times and continuously adds the work dir if it is not an absolute path. The modification of the config variable here looks strange in any case.

Reply:
The more general problem is that the `cs.` obsolete legacy structure is "abused" for the new-style config. This makes things pretty complicated. Would best be refactored as well.

Reply:
Thanks for suggestion, will try to not forget it in next commit.